js/src/forum/components/CategoriesPage.tsx

```diff
@@ -56,18 +56,38 @@ export default class CategoriesPage extends Page {
 
     items.add('hero', IndexPage.prototype.hero(), 100);
 
+    items.add('container', this.containerItems().toArray(), 50);
+
+    return items;
+  }
+
+  containerItems() {
+    const items = new ItemList();
+    const indexPage = IndexPage.prototype.view();
     items.add(
       'container',
       <div className={app.forum.attribute('categories.fullPageDesktop') ? 'container topNavContainer' : 'container sideNavContainer'}>
-        {this.containerItems().toArray()}
+        {this.contentItems().toArray()}
       </div>,
       50
     );
-
+    // Only check for widget for the header and footer if enable in the settings
+    if ((app.forum.attribute('categories.widgetHeader') || app.forum.attribute('categories.widgetFooter')) && indexPage.children.length > 1 && indexPage.children[1].children) {
+      indexPage.children[1].children.forEach((child, index) => {
```

Comment:
Why is this happening in a loop? Why not use `children.some()` to determine whether there exists a child that would trigger the widgets to be added? And optional chaining to simplify the if statement?

That being said, this code is very brittle: structure of vdom nodes is subject to change, and relying on the absence of a class name is prone to go wrong as soon as someone sees one.

Is there some value / better API we could expose in Core that would make this more reliable?

Reply:
I will work on a better detection on this later

Reply:
I haven't learnt of a way to grab the non populated extended in a super reliable way. I have made a better method in going through known classname instead and look for undefined className because they are from the widget extension. That way the extension can search through the tiers of containers inside the IndexPage.